File: scala/leap/src/main/scala/Leap.scala
object Leap {
  def leapYear(year: Int): Boolean = {
    if (year % 4 == 0) {
      if (year % 100 == 0) {
        return (year % 400 == 0)
      } else {
        return true
      }
    }
    false
  }
}

File: clojure/leap/src/leap.clj
(ns leap)

(defn divisible? [x y]
  (zero? (mod x y))
  )

(defn leap-year? [year]
  ;; curried function
  (defn yeardivisible? [x]
    (divisible? year x)
    )

  (if (yeardivisible? 4)
    (or (not (yeardivisible? 100))
        (yeardivisible? 400))
    false)
  )
